### PR TITLE
Fix http server timeouts documentation

### PR DESCRIPTION
**Pull request by croudet**
~~remove unused readTimeout~~

### CODE REVIEW

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java [hunk 1]
```diff
@@ -282,15 +273,6 @@ public void setHost(String host) {
...
 
-    /**
-     * Sets the default read timeout.
-     *
-     * @param readTimeout The read timeout
-     */
-    public void setReadTimeout(Integer readTimeout) {
-        this.readTimeout = readTimeout;
-    }
-
```
Comment (jameskleeh) [inline http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: This is a breaking change and cannot be done in a 1.x release

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java [hunk 2]
```diff
@@ -71,13 +70,13 @@
      * The default read idle time in minutes.
      */
     @SuppressWarnings("WeakerAccess")
-    public static final long DEFAULT_READ_IDLE_TIME_MINUTES = 5;
+    public static final Duration DEFAULT_READ_IDLE_TIME_MINUTES = Duration.ofMinutes(5L);
```
Comment (jameskleeh) [inline http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: This is a breaking change

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java [hunk 3]
```diff
@@ -71,13 +70,13 @@
...
 
     /**
      * The default write idle time in minutes.
      */
     @SuppressWarnings("WeakerAccess")
-    public static final long DEFAULT_WRITE_IDLE_TIME_MINUTES = 5;
+    public static final Duration DEFAULT_WRITE_IDLE_TIME_MINUTES = DEFAULT_READ_IDLE_TIME_MINUTES;
```
Comment (jameskleeh) [inline http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: This is a breaking change

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java [hunk 4]
```diff
@@ -163,13 +161,6 @@ public Charset getDefaultCharset() {
         return Optional.ofNullable(host);
     }
 
-    /**
-     * @return The read timeout setting for the server
-     */
-    public Optional<Integer> getReadTimeout() {
-        return Optional.ofNullable(readTimeout);
-    }
-
```
Comment (jameskleeh) [inline http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: This is a breaking change

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java (hunk 1 above)
Comment (croudet) [reply http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: Reverted changes.

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java (hunk 3 above)
Comment (croudet) [reply http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: Reverted changes.

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java (hunk 4 above)
Comment (croudet) [reply http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: Reverted changes.

File: http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java (hunk 2 above)
Comment (croudet) [reply http-server/src/main/java/io/micronaut/http/server/HttpServerConfiguration.java]: Reverted changes.

**Pull request merged**